shared/js/dataDriver.js

```diff
@@ -149,6 +152,13 @@ loop.DataDriver = function() {
             console.log(`Error handling EventSource.put: ${event.data}`, ex);
           }
           break;
+
+        // Handle an error, either unable to initially connect or
+        // unexpected disconnection
+        case "error":
```

Comment:
Switched to only reconnect when readyState === CLOSED. The way the spec is written, we shouldn't ever have to manually reconnect, eventsource should take care of reconnecting in all recoverable situations. However, chrome and firefox have different behaviors that in practice means reconnecting on error when readyState === CLOSED does help for some cases in Firefox. Chrome doesn't ever need the manual code and always correctly reconnects eventually in case of network failure.

However, there are some cases when Firefox says readyState === CONNECTING, but never connects, never sends another error event, and never successfully connects. I will test for longer to see if it eventually recovers, but I'm not really sure what we can do about this case other than file a bug against gecko.